README.tfdoc.hcl

```diff
@@ -0,0 +1,932 @@
+                END
+              }
+            }
+
+            attribute "transition" {
+              type        = object(transition)
+              description = <<-END
+                Specifies a period in the object's transitions (documented below).
+              END
+
+              attribute "date" {
+                type        = string
+                description = <<-END
+                  Specifies the date after which you want the corresponding action to take effect.
+                END
+              }
+
+              attribute "days" {
+                type        = number
+                description = <<-END
+                  Specifies the number of days after object creation when the specific rule action takes effect.
+                END
+              }
+
+              attribute "storage_class" {
+                required    = true
+                type        = string
+                description = <<-END
+                  Specifies the Amazon S3 storage class to which you want the object to transition.
+                  Can be `ONEZONE_IA`, `STANDARD_IA`, `INTELLIGENT_TIERING`, `GLACIER`, or `DEEP_ARCHIVE`.
+                END
+              }
+            }
+
+            attribute "noncurrent_version_expiration" {
+              type        = object(noncurrent_version_expiration)
+              description = <<-END
+                Specifies when noncurrent object versions expire (documented below).
+              END
+
+              attribute "days" {
+                required    = true
+                type        = number
+                description = <<-END
+                  Specifies the number of days an object's noncurrent object versions expire.
+                END
+              }
+            }
+
+            attribute "noncurrent_version_transition" {
+              type        = object(noncurrent_version_transition)
+              description = <<-END
+                Specifies when noncurrent object versions transitions (documented below).
+                At least one of `expiration`, `transition`, `noncurrent_version_expiration`, `noncurrent_version_transition` must be specified.
+              END
+
+              attribute "days" {
+                required    = true
+                type        = number
+                description = <<-END
+                  Specifies the number of days an object's noncurrent object versions expire.
+                END
+              }
+
+              attribute "storage_class" {
+                required    = true
+                type        = string
+                description = <<-END
+                  Specifies the Amazon S3 storage class to which you want the noncurrent versions object to transition.
+                  Can be `ONEZONE_IA`, `STANDARD_IA`, `INTELLIGENT_TIERING`, `GLACIER`, or `DEEP_ARCHIVE`.
+                END
+              }
+            }
+          }
+        }
+
+        section {
```

Comment:
this and following subsections should go below extended resource configurations..